installer/manifests/keptn/templates/_helpers.tpl

```diff
@@ -71,3 +71,47 @@ resources:
     memory: "32Mi"
     cpu: "100m"
 {{- end }}
+
+{{/*
+Renders nodeSelector if either "value" or "default" is not empty. the needed indentation must be set
+with "indent". Templates may be used in values and keys!
```

Comment:
Is there a reason why you put the indentation into the helper function instead of using it without it and then using the `nindent` or `indent` function outside?

then instead of this:
```
{{- include "keptn.nodeSelector" (dict "value" .Values.statisticsService.nodeSelector "default" .Values.common.nodeSelector "indent" 6 "context" . )}}
```
you would use:
```
{{- include "keptn.nodeSelector" (dict "value" .Values.statisticsService.nodeSelector "default" .Values.common.nodeSelector "context" . ) | indent 6 }}
```

I am not fully sure if this would actually work though...

Reply:
The reason behind is, that i wanted to avoid all additional/senseless whitespace. 

The template
```
      terminationGracePeriodSeconds: 30
      {{- include "keptn.nodeSelector" (dict "value" .Values.statisticsService.nodeSelector "default" .Values.common.nodeSelector "context" . ) | indent 6 }}
```
with unset nodeSelector renders to
```
      terminationGracePeriodSeconds: 30< 6 Spaces>   
```

You could also avoid that with an if around the include (but i also wanted to avoid duplicate code, because that "if" would be needed for each deployment)

I think that all 3 solutions are completely valid! 

1. current solution (avoids whitespace but introduces `indent` parameter for template function)
2. applying `indent` outside template function (renders 6 space chars when no nodeSelector is set)
3. solution 2 + additional if around template function call (no whitespace but more code)

I would be happy to adapt the code to the preferred solution.

Reply:
In that I would say you already chose the best solution :)
Let's leave it the way it is. That will result in the cleanest helm template and also less code to maintain